fl4health/processing/VAE_dim_reduction.py

```diff
@@ -0,0 +1,87 @@
+from abc import ABC
+from pathlib import Path
+from typing import Tuple
+
+import numpy as np
+import torch
+
+from fl4health.model_bases.autoencoders_base import ConditionalVAE, VarioationalAE
+
+
+class Processing(ABC):
+    """
+    Abstact class to add processing methods.
```

Comment:
I understand that VAE and CVAE may offer the better representations but since we also support Basic_AE, would it be good to also have Basic_AE processor?

Reply:
I agree, will add that.

Reply:
I feel all of these three functions are so similar, can we handle it in `auto encoder_trainer.py` and have just one function? As we also have a different CVAE_trainer.

Reply:
Yeah, I think the functionality of the Processor module is pretty thin and it may be best to just include this loading in the trainer to have all the functionality in one place to simplify things. @sanaAyrml also raises a good point that we already have separate classes anyway.

Reply:
This is a perfect suggestion! I pushed a new design based on the comments. Hopefully, this has increased the readability, and decreased code complexity.